src/kernel/include/shared/elf_loader.h

```diff
@@ -65,34 +65,26 @@ elf_chk_format(struct elf_hdr *hdr)
 	unsigned char *s = hdr->e_ident;
 
 	/* Format:  "0x7fELF1" where the 1 is for 32 bit */
-	if (s[0] != 0x7f || s[1] != 'E' || s[2] != 'L' || s[3] != 'F' || s[4] != 1) return -1;
-	return 0;
-}
-
-/*
- * Returns the address of the entry function, or 0 on error.
- */
-static inline vaddr_t
-elf_entry_addr(struct elf_hdr *hdr)
-{
-	if (elf_chk_format(hdr)) return 0;
+	if (s[0] == 0x7f && s[1] == 'E' && s[2] == 'L' && s[3] == 'F' && s[4] == 1) return 1;
```

Comment:
I don't like the magic values of 1 and 2, but didn't have a better idea. So I didn't comment on it last time.

But here's an idea: how about return `32` or `64` accordingly! At least then the return values hold some reasonable semantic meaning that is easy to remember.